.github/workflows/ci-pipeline.yml

```diff
@@ -98,7 +98,17 @@ jobs:
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v3
 
-      - name: Build Docker image
+      - name: Azure Login
+        if: github.ref == 'refs/heads/main' && github.event_name == 'push'
+        uses: azure/login@v2
+        with:
+          creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+      - name: Login to Azure Container Registry
+        if: github.ref == 'refs/heads/main' && github.event_name == 'push'
+        run: az acr login --name ${{ vars.ACR_NAME }}
+
+      - name: Build Docker image (PR)
```

Comment:
I would move azure login and container registry login after you've **built** the image, since you don't need it here.

You can add this back just before  - name: Push to Azure Container Registry and make all of those only run on master branch :)